packages/icons/lib/build-worker-utils.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * External dependencies
+ */
+const path = require( 'path' );
+const { promisify } = require( 'util' );
+const execFile = promisify( require( 'child_process' ).execFile );
+
+/**
+ * Internal dependencies
+ */
+const { generateTsxFiles } = require( './generate-library' );
+
+const ICON_LIBRARY_DIR = path.join( __dirname, '..', 'src', 'library' );
+
+/**
+ * A build-worker task to be used by the monorepo's watcher-builder.
+ *
+ * @see bin/packages/build-worker.js
+ *
+ * @param {string} file File to build.
+ */
+async function buildSVG( file ) {
+	if ( path.dirname( file ) !== ICON_LIBRARY_DIR ) {
+		return false;
+	}
+
+	try {
+		await execFile( 'git', [ 'ls-files', '--error-unmatch', file ] );
+	} catch {
+		throw new Error(
+			`Cannot generate icon from untracked SVG file '${ path.basename(
+				file
+			) }'.
+Please add it to Git, then restart:
+
+	git add ${ path.relative( '', file ) }
+	npm run dev
+`
+		);
+	}
```

Comment:
```suggestion
	const FgRed = '\x1b[31m';
	const Reset = '\x1b[0m';

	try {
		await execFile( 'git', [ 'ls-files', '--error-unmatch', file ] );
	} catch {
		throw new Error(
			`${ FgRed }Cannot generate icon from untracked SVG file '${ path.basename(
				file
			) }'.${ Reset }
${ FgRed }Please add it to Git, then restart:${ Reset }

${ FgRed }	git add ${ path.relative( '', file ) }${ Reset }
${ FgRed }	npm run dev${ Reset }
`
		);
	}
```

Nit: What do you think about making this message red to make it easier to see that an error has occurred?

<img width="1272" height="584" alt="Image" src="https://github.com/user-attachments/assets/58a6f319-7183-4829-a684-29f14a77399b" />

Reply:
I don't feel strongly about it :) 

* Argument against: red isn't used in the watcher for other failures, like parse errors, which are just as important.
* Argument in favour: this message is different in that it is a set of instructions for the developer, so it should look different.

I'll apply it, thanks!